airflow/jobs/job.py

```diff
@@ -292,3 +265,181 @@ def most_recent_job(job_type: str, session: Session = NEW_SESSION) -> Job | None
         )
         .first()
     )
+
+
+@provide_session
+def run_job(job: Job | JobPydantic, session: Session = NEW_SESSION) -> int | None:
+    """
+    Runs the job. The Job is always an ORM object and setting the state is happening within the
+    same DB session and the session is kept open throughout the whole execution
+
+    :meta private:
+
+    TODO: Maybe we should not keep the session during job execution ?.
+    """
+    if _ENABLE_AIP_44:
+        return _run_job_for_AIP_44(job=job, session=session)
```

Comment:
More concretely, what I mean is instead of adding these separate run, prepare, and complete functions, keep it simple and just have this block fetch the db job and follow the normal path for now. When we are ready to not use the db, we can review the structure then.

Reply:
Yeah, I agree the way it is now is a bit "ugly" and I agree it is in parts YAGNI case

But I am flexible. I think I  can actually remove all the coniditional code here and rebase the last step of refactor #30376 as now I know how the final stage will be best (and we can bring back AIP-44 changes as next steps after 2.6 cut-off.

Reply:
There you go. 

I removed the AIP-44 code now. Rebasing and removing it from #30376 now.